core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -666,10 +667,20 @@ object Validator {
         val objectFields    = fields(obj)
         val supertypeFields = supertype.flatMap(fields)
 
+        def isNonNullableSubtype(supertypeFieldType: __Type, objectFieldType: __Type) = {
```

Comment:
I suspect this is a bug that the tests didn't test (although it seemed like they tested it).